packages/apollo-voyager-audit/src/index.ts

```diff
@@ -0,0 +1,75 @@
+import { buildPath } from '@aerogear/apollo-voyager-tools'
+import { GraphQLResolveInfo } from 'graphql'
+import { IFieldResolver } from 'graphql-tools'
+import pino from 'pino'
+
+const log = pino()
+const auditLogger = log.child({tag: 'AUDIT'})
+
+let auditLogEnabled = false
+
+export function isAuditLogEnabled (): boolean {
+  return auditLogEnabled
+}
+
+export function setAuditLogEnabled (val: boolean): void {
+  auditLogEnabled = val
+}
+
+interface ResolverObject {
+  [key: string]: IFieldResolver<any, any>
+}
+
+export function wrapResolversForAuditLogging (resolverMappings: { [key: string]: ResolverObject }): { [key: string]: ResolverObject } {
+  const output: { [key: string]: ResolverObject } = {}
+
+  const typeKeys = Object.keys(resolverMappings)
+  for (const typeKey of typeKeys) {
+    output[typeKey] = {}
+
+    const fieldResolversForType = resolverMappings[typeKey]
+    const fieldKeysForType = Object.keys(fieldResolversForType)
+    for (const fieldKey of fieldKeysForType) {
+      const resolverForField = fieldResolversForType[fieldKey]
+      output[typeKey][fieldKey] = wrapSingleResolverForAuditLogging(resolverForField)
+    }
+  }
+
+  return output
+}
+
+export function auditLog (success: boolean, request: any, info: GraphQLResolveInfo, parent: any, args: any, msg: string) {
+  if (auditLogEnabled) {
+    auditLogger.info({
```

Comment:
[Optional] Define this as interface so we have automatically document in the code what is being logged and when.